### ISSUE TITLE
Define supported ABI

### ISSUE THREAD

**Issue by krasin** [llvm-dcpu16/llvm-dcpu16#69]
Currently, the implementation uses an ad-hoc ABI which is not fully consistent and sometimes derived from MSP430.

We need to explicitly define the supported ABI. Sure, it's the subject to change because Notch will hopefully allow to address values on stack with [4+SP].

From what I remember:
- We use SP just to push and pop ret addresses (is it still true?)
- We use C as a pointer to data stack (kind of frame pointer)
- We pass argument in X, Y, Z (but probably on other registers too, need to look at the implementation)
- We return values in X, Y, Z (usually, just X is used)

**Comment by ghost** [comment]
The workaround with the C register should be removed when notch has hopefully updated his specification. I allready tested one version with only the SP register, without the help of C and it produced code which locked "okay".

There has been some work in the "community" for an ABI recently:
http://0x10cforum.com/forum/m/4932880/viewthread/2720680-important-ongoing-standardisation-0x10c-dcpu16-virtual-cpu

**Comment by krasin** [comment]
Yeah, I saw that ABI. Clearly, we need to wait few days for updates from Notch before stabilizing (or pretending to do so) the ABI.
